lib/moon/design/modal.ex

```diff
@@ -1,33 +1,56 @@
 defmodule Moon.Design.Modal do
   @moduledoc false
-  use Moon.StatelessComponent
+  use Moon.StatefulComponent
+
+  data(is_open, :boolean, default: false)
+  data(is_closing, :boolean, default: false)
```

Comment:
maybe it should be prop, so user can make it visible by default. the same for a drawer